### ISSUE TITLE
Cannot initialize database with indexes in API 15 (ICS)

### ISSUE THREAD

**Issue by monzee** [yahoo/squidb#51]
When the `getIndexes` method is overriden in the `AbstractDatabase` impl, the SQLite helper throws:
```
     Caused by: java.lang.IllegalStateException: getWritableDatabase called recursively
            at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:140)
            at com.yahoo.squidb.data.AbstractDatabase.openForWriting(AbstractDatabase.java:370)
            at com.yahoo.squidb.data.AbstractDatabase.getDatabase(AbstractDatabase.java:284)
            at com.yahoo.squidb.data.AbstractDatabase.tryExecSql(AbstractDatabase.java:1028)
            at com.yahoo.squidb.data.AbstractDatabase.tryCreateIndex(AbstractDatabase.java:984)
            at com.yahoo.squidb.data.AbstractDatabase.tryCreateIndex(AbstractDatabase.java:955)
            at com.yahoo.squidb.data.AbstractDatabase$DatabaseHelper.onCreate(AbstractDatabase.java:804)
            at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:165)
            at com.yahoo.squidb.data.AbstractDatabase.openForWriting(AbstractDatabase.java:370)
            at com.yahoo.squidb.data.AbstractDatabase.getDatabase(AbstractDatabase.java:284)
            at com.yahoo.squidb.data.AbstractDatabase.rawQuery(AbstractDatabase.java:469)
            at com.yahoo.squidb.data.DatabaseDao.query(DatabaseDao.java:92)
            at com.yahoo.squidb.data.DatabaseDao.fetchFirstItem(DatabaseDao.java:596)
            at com.yahoo.squidb.data.DatabaseDao.fetchFirstItem(DatabaseDao.java:589)
            at com.yahoo.squidb.data.DatabaseDao.fetchByCriterion(DatabaseDao.java:121)
           (my code from here)
```
If I return `null` from `getIndexes` the error goes away (but then I have no indexes). I haven't tried to find a workaround yet, but maybe creating indexes manually in `onTablesCreated` would work.

**Comment by jdkoren** [comment]
Thanks for reporting this, I am working on the fix now

**Comment by jdkoren** [comment]
Fixed in https://github.com/yahoo/squidb/pull/54

**Comment by sbosley** [comment]
The fix for this was released in our launch of 1.1.0 today.

**Comment by inawlaljar** [comment]
Is it possible to create index on multiple columns? 

for e.g. 
 tryCreateIndex(User.TABLE.index("ss_fb_def_item_idx", [User.ADDRESS_ID, User.ORG_ID, User.EDUCATION_ID]));
